SwordImportExportPlugin.inc.php

```diff
@@ -0,0 +1,230 @@
+<?php 
+
+/**
+ * @file plugins/generic/sword/SwordImportExportPlugin.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2003-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class SwordImportExportPlugin
+ * @ingroup plugins_generic_sword
+ *
+ * @brief Sword deposit plugin
+ */
+
+import('lib.pkp.classes.plugins.ImportExportPlugin');
+
+class SwordImportExportPlugin extends ImportExportPlugin {
+	/** @var SwordPlugin Parent plugin */
+	protected $_parentPlugin = null; 
+
+	/**
+	 * Constructor
+	 * @param $parentPlugin SwordPlugin
+	 * 
+	 */
+	public function __construct($parentPlugin) {
+		parent::__construct();
+		$this->_parentPlugin = $parentPlugin;
+	}
+
+	/**
+	 * Called as a plugin is registered to the registry
+	 * @param $category String Name of category plugin was registered to
+	 * @return boolean True if plugin initialized successfully; if false,
+	 * 	the plugin will not be registered.
+	 */
+	public function register($category, $path) {
+		$success = parent::register($category, $path);
+		$this->addLocaleData();
+		return $success;
+	}
+
+	/**
+	 * Get reference to the sword plugin
+	 * @return SwordPlugin
+	 */
+	public function getSwordPlugin() {
+		return $this->_parentPlugin;
+	}
+
+	/**
+	 * Get the name of this plugin. The name must be unique within
+	 * its category.
+	 * @return String name of plugin
+	 */
+	public function getName() {
+		return 'SwordImportExportPlugin';
+	}
+
+	/**
+	 * Get the display name.
+	 * @return string
+	 */
+	public function getDisplayName() {
+		return __('plugins.importexport.sword.displayName');
+	}
+
+	/**
+	 * Get the display description.
+	 * @return string
+	 */
+	public function getDescription() {
+		return __('plugins.importexport.sword.description');
+	}
+
+	/**
+	 * @copydoc Plugin::getTemplatePath($inCore)
+	 */
+	public function getTemplatePath($inCore = false) {
+		return $this->getSwordPlugin()->getTemplatePath($inCore);
+	}
+
+	/**
+	 * Display the plugin.
+	 * @param $args array
+	 * @param $request PKPRequest
+	 */
+	public function display($args, $request) {
+		parent::display($args, $request);
+		$templateMgr = TemplateManager::getManager($request);
+		$context = $request->getContext();
+		$journal = $request->getJournal();
```

Comment:
`$request->getContext` and `$request->getJournal` are the same thing.

Reply:
`$journal = $request->getJournal();` removed.